src/plugins/charts/public/services/palettes/helpers.ts

```diff
@@ -20,7 +21,7 @@ function findColorSegment(
 
   // what about values in range
   const index = colors.findIndex((c, i) => comparison(value, rangeMin + (1 + i) * step) <= 0);
-  return colors[index] || colors[0];
+  return colors[index] || colors[colors.length - 1];
```

Comment:
The fallback here covers the case where `index = -1`. Why picking the last value in this case?